File: backend/src/main/java/net/ugurkartal/backend/services/abstracts/CategoryService.java
package net.ugurkartal.backend.services.abstracts;

import net.ugurkartal.backend.services.dtos.requests.CategoryRequest;
import net.ugurkartal.backend.services.dtos.responses.CategoryCreatedResponse;
import net.ugurkartal.backend.services.dtos.responses.CategoryGetAllResponse;

import java.util.List;

public interface CategoryService {
    List<CategoryGetAllResponse> getAllCategories();
    CategoryCreatedResponse getCategoryById(String id);
    CategoryCreatedResponse addCategory(CategoryRequest categoryRequest);
}

File: frontend/src/services/CategoryService.ts
import axios from "axios";
import {Category} from "../types/Category.ts";

export default class CategoryService {
    getAllCategories() {
        return axios.get('/api/categories');
    }

    getCategoryById(id: string) {
        return axios.get(`/api/categories/${id}`);
    }

    addCategory(category: Category) {
        return axios.post('/api/categories', category);
    }
}